kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -150,85 +150,123 @@ define([
             };
         }
 
-        function getObjectInfo(workspaceId, objectName, authToken, serviceUrl) {
+        /**
+         *
+         * @param {number} workspaceId
+         * @param {Array<string>} objectNames
+         * @param {string} authToken
+         * @param {string} serviceUrl
+         * @returns Promise resolving into an Array of strings, in the same order as the given objectNames
+         */
+        function getObjectTypes(workspaceId, objectNames, authToken, serviceUrl) {
```

Comment:
Changes here all all to create an Array version of the workspace object name validator.